File: LeetCode_每日一题/2023-06/07_2611. 老鼠和奶酪.go
/*
 * @Author: hakusai
 * @Date: 2023-06-07 23:26:40
 * @LastEditTime: 2023-06-07 23:43:41
 * @Description:
 */
package main

import "sort"

func miceAndCheese(reward1 []int, reward2 []int, k int) (ans int) {
	for i, x := range reward2 {
		ans += x
		reward1[i] -= x
	}
	sort.Ints(reward1)
	n := len(reward1)
	for i := 0; i < k; i++ {
		ans += reward1[n-i-1]
	}
	return
}

File: LeetCode_每日一题/2023-06/07_2611. 老鼠和奶酪.java

/*
 * @Author: hakusai
 * @Date: 2023-06-07 23:26:35
 * @LastEditTime: 2023-06-07 23:41:08
 * @Description: 
 */
import java.util.*;
import java.io.*;
import java.math.*;

class Solution {
    public int miceAndCheese(int[] reward1, int[] reward2, int k) {
        int ans = 0;
        int n = reward1.length;
        for (int i = 0; i < n; i++) {
            ans += reward2[i];
            reward1[i] -= reward2[i];
        }
        Arrays.sort(reward1);
        for (int i = 0; i < k; i++) {
            ans += reward1[n - i - 1];
        }
        return ans;
    }
}